src/components/core/core.js

```diff
@@ -308,13 +308,17 @@ export default class Core extends UIObject {
     return this.activeContainer && this.activeContainer.getPlaybackType()
   }
 
+  isFullscreen() {
+    return Fullscreen.getFullscreenElement() === (Browser.isiOS ? this.activeContainer.el : this.el)
+  }
+
   toggleFullscreen() {
-    if (!Fullscreen.isFullscreen()) {
-      Fullscreen.requestFullscreen(Browser.isiOS ? this.activeContainer.el : this.el)
-      !Browser.isiOS && this.$el.addClass('fullscreen')
-    } else {
+    if (this.isFullscreen()) {
```

Comment:
@towerz Do you think that this validation should occur in both places (plugin and core)?
The plugin doesn't actually need to `toggleFullscreen`, it always want to cancel it.

Reply:
@luanraithz the validation should be in both because they aren't due to the same reason. In the Core `toggleFullscreen` the check is necessary to determine whether to enter or exit fullscreen.

In the case of the plugin, the check is necessary to avoid an unnecessary call to `cancelFullscreen`.